expose-utility-functions-to-states.md

```diff
@@ -0,0 +1,133 @@
+- Feature Name: Expose utility functions to states, but not to the CLI or template context
+- Start Date: 2023-06-06
+- SEP Status: Draft
+- SEP PR: https://github.com/saltstack/salt-enhancement-proposals/pull/70
+- Salt Issue: https://github.com/saltstack/salt/pull/64287#pullrequestreview-1451646261
+
+# Summary
+[summary]: #summary
+
+Provide a means for the loader to access utility functions, but keep them from
+being added to the CLI or template context.
+
+# Motivation
+[motivation]: #motivation
+
+It is a policy of the Salt core team not to add any new utility functions to
+the `__salt__` loader (thus exposing them to the Salt CLI and the template
+context). The core team's proposed alternative, however, is to do an OS check
+in the state module, then a late import of the salt execution module, and
+finally to invoke the utility function directly. While this option technically
+works, it is unequivocally a bad idea, with several complications:
+
+1. The state modules were never supposed to include platform-specific code. I
+   was even [advised to
+   remove](https://github.com/saltstack/salt/pull/3019#issuecomment-11680999)
+   grains checks from a state module some 10 years ago.
```

Comment:
Let me start by pointing out that 10 years is a long time ago, things have changed. The main idea, `never supposed to include platform-specific code` is still valid, **when possible**.

Reply:
When not possible, there are still ways to achieve it without code duplication.

Reply:
Then name them?

Reply:
It was named

https://github.com/saltstack/salt-enhancement-proposals/pull/70#discussion_r1229320359